review of cp

mikey2020 opened the pull request.

Commit:
install node-sass

Commit:
Design templates with material design lite

Commit:
Write failing tests for routes

Commit:
Write one more test for routes

Commit:
Update webpack config file for production

Commit:
Update webpack config file for production

Commit:
Update webpack config file for production

Commit:
Update webpack config file for production

Commit:
Update webpack config file for production

Commit:
Update webpack config file for production

Commit:
Update webpack config file for production

Commit:
Update webpack config file for production

Commit:
Update webpack config file for production

Commit:
Update webpack config file for production

Commit:
Update webpack config file for production

Commit:
Update webpack config file for production

Commit:
Update webpack config file for production

Commit:
Update gulpfile

Commit:
Update gulpfile for coveralls

Commit:
Change gulplif.js to gulpfile.babel.js

Commit:
Update gulpfile.babel.js

Commit:
Modify gulpfile

Commit:
Update app.js file

Commit:
Add coveralls module for coveralls.io

Commit:
Use node coveralls to send coverage

Commit:
Update travis file to run coverage

Commit:
Update travis file to run coverage

Commit:
Update travis file to run coverage

Commit:
Update travis file to run coverage

Commit:
Update coveralls.yml file

Commit:
Update coveralls.yml file

Commit:
Update coveralls.yml file

Commit:
Update coveralls.yml file

Commit:
Update coveralls.yml file

Commit:
Update coveralls.yml file

Commit:
Update coveralls.yml file

Commit:
Update coveralls.yml file

Commit:
Update coveralls.yml file

Commit:
Refactor code to pass eslint tests

Commit:
Refactor code to pass eslint tests

Commit:
Refactor code to pass eslint tests

Commit:
Refactor code to pass eslint tests

Commit:
Refactor code to pass eslint tests

Commit:
Update README.md file

Commit:
Write tests for validations

Commit:
Write tests for functions

mikey2020 committed:
Update README.md

Commit:
Fix style errors

Commit:
Fix some eslint errors

Commit:
Update read me file

Commit:
Fix some eslint errors

Commit:
Fix some eslint errors

Commit:
Fix some eslint errors

Commit:
Add empty client folder

Commit:
Fix eslint errors

Commit:
Set up development environment on mac

Commit:
Remove package-lock.json

Commit:
Update package.json file

Commit:
Add package-lock.json to origin

Commit:
Remove package-lock.json

mikey2020 committed:
Add more routes test

mikey2020 committed:
Run coveralls

mikey2020 committed:
Create .codeclimate.yml file

mikey2020 committed:
Fix style guide errors in groupcontroller file

mikey2020 committed:
Fix merge conflict in groupcontroller file

mikey2020 committed:
Fix eslint errors in groupcontroller.js

mikey2020 committed:
Fix eslint errors in routesTest.js

mikey2020 committed:
Remove coverage,dist folders from remote repository

mikey2020 committed:
Change codeclimate.yml file

mikey2020 committed:
Change codeclimate.yml file

mikey2020 committed:
Fix eslint errors in validations

mikey2020 committed:
Fix eslint errors in models , groupcontroller files

mikey2020 committed:
Update read me

mikey2020 committed:
Add passwordConfirmation information to read me file

mikey2020 committed:
Fix heroku url mistake on readme

mikey2020 committed:
Fix heroku url mistake on readme

mikey2020 committed:
Update read me file

mikey2020 committed:
Update read me file

mikey2020 committed:
Update README.md

mikey2020 committed:
Use 'this' in usercontroller class

mikey2020 committed:
Add code to make sure only valid user is added to a group

mikey2020 committed:
Add validations for edge cases

mikey2020 committed:
Create index page for app

mikey2020 committed:
Validate uniqueness of all user's input

mikey2020 committed:
Validate uniqueness of all user's input

mikey2020 committed:
Write test for  signin input validation

mikey2020 committed:
Remove duplications

mikey2020 committed:
Fix eslint errors in db.js file

mikey2020 committed:
Remove webpack.config.js

mikey2020 committed:
Fix sequelize error

mikey2020 committed:
Remove duplictions from validations

mikey2020 committed:
Remove duplictions from validations

mikey2020 committed:
Update README.md

mikey2020 committed:
Update codeclimate.yml file

mikey2020 committed:
Fix merge errors

mikey2020 committed:
Update codeclimate.yml file

mikey2020 committed:
Update codeclimate.yml file

mikey2020 committed:
Convert database to use migrations

mikey2020 committed:
Use environment variables

mikey2020 committed:
Update config.json file

mikey2020 committed:
Update config.json file

mikey2020 committed:
Update readme file

mikey2020 committed:
Fix merge conflict

mikey2020 committed:
Update readme file

mikey2020 committed:
Update readme file

mikey2020 committed:
Update readme file

mikey2020 committed:
Update readme file

mikey2020 committed:
Create index.html

mikey2020 committed:
Allow user to test routes with postman

mikey2020 committed:
Make adding user to a group unique

mikey2020 committed:
Make proper validations

mikey2020 committed:
Make proper validations

mikey2020 committed:
Make proper validations

mikey2020 committed:
Fix eslint errors

mikey2020 committed:
Edit login.html file:

mikey2020 committed:
Improve ui/ux design

mikey2020 committed:
Improve templates

mikey2020 committed:
Validate routes

mikey2020 committed:
Validate routes

mikey2020 committed:
Update local with remote

mikey2020 committed:
Add method to check group memebers

mikey2020 committed:
Fix merge conflicts

mikey2020 committed:
Make sure all tests pass

mikey2020 committed:
(feat): Setup basic navigation with react-router

mikey2020 committed:
Add addUser action to signupActions.js

mikey2020 committed:
(feat): style landing page for post it

mikey2020 committed:
Set up sign in and sign up with react-redux

mikey2020 committed:
Make sure branch is up to date with origin

mikey2020 committed:
Remove models.js and update other models

mikey2020 committed:
Refactor to pass tests

mikey2020 committed:
Update travis.yml

mikey2020 committed:
Update config.json file

mikey2020 committed:
Update groupcontroller

mikey2020 committed:
Update routes test file

mikey2020 committed:
Update middlewares

mikey2020 committed:
Update usercontroller file

mikey2020 committed:
Update config.json file

mikey2020 committed:
Update routes test

mikey2020 committed:
Update travis.yml file

mikey2020 committed:
Update travis.yml file

mikey2020 committed:
Refactor tests and Update travis.yml

mikey2020 committed:
Update travis.yml

mikey2020 committed:
Update travis.yml

mikey2020 committed:
Update travis.yml

mikey2020 committed:
Update package.json and travis.yml

mikey2020 committed:
Update package.json and travis.yml

mikey2020 committed:
Update package.json and travis.yml

mikey2020 committed:
Update package.json and travis.yml

mikey2020 committed:
Update package.json and travis.yml

mikey2020 committed:
Update package.json and travis.yml

mikey2020 committed:
Update package.json and travis.yml

mikey2020 committed:
Update models/index.js file

mikey2020 committed:
Update travis.yml

mikey2020 committed:
Update package.json

mikey2020 committed:
Update package.json

mikey2020 committed:
Setup signup and signin in react-redux

mikey2020 committed:
Update signup validations

mikey2020 committed:
Delete db.js file

mikey2020 committed:
Update package.json file

mikey2020 committed:
Fix merge conflicts

mikey2020 committed:
Remove package-lock.json

mikey2020 committed:
Remove package-lock.json

mikey2020 committed:
Fix webpck plus sequelize error

mikey2020 committed:
Add DS_strore

mikey2020 committed:
Update .travis.yml and npm scripts

mikey2020 committed:
Add model tests

mikey2020 committed:
Set up signin form component

mikey2020 committed:
Setup redux testing with jest

mikey2020 committed:
Setup react and redux testing

mikey2020 committed:
Update package.json for easy test running

mikey2020 committed:
Setup homepage functionalities

mikey2020 committed:
Set up adding ('not finished')

mikey2020 committed:
Set up feature 'users should be able to add other users'

mikey2020 committed:
Setup production build for react

mikey2020 committed:
Add signin form test

mikey2020 committed:
Set up jwt token generation

mikey2020 committed:
Write more component tests

mikey2020 committed:
feat/write component tests

mikey2020 committed:
Setup jwt for user authentication

mikey2020 committed:
Add priority column in message model

mikey2020 committed:
Re-design signup page

mikey2020 committed:
Design components

mikey2020 committed:
Make all tests pass

mikey2020 committed:
Update travis.yml

mikey2020 committed:
Update config.json file

mikey2020 committed:
Update travis.yml

mikey2020 committed:
Update dist folder

mikey2020 committed:
Refactor code to pass tests

mikey2020 committed:
Fix merge conflicts

mikey2020 committed:
Fix merge conflicts

mikey2020 committed:
Add validations file in client folder

mikey2020 committed:
Set up jwt on client ande server sides

mikey2020 committed:
Exclude dist folder from codeclimate.yml

mikey2020 committed:
Create pagination component

mikey2020 committed:
Change flash message design

mikey2020 committed:
Add static method reset password

mikey2020 committed:
Add reset password feature

mikey2020 committed:
Setup feature user should other user

mikey2020 committed:
Set footer to be fixed on page

mikey2020 committed:
Change module in webpack.config.dev.js

mikey2020 committed:
Update package.json with testing modules

mikey2020 committed:
Make production work weel

mikey2020 committed:
Fix merge conflicts

mikey2020 committed:
Set up updated production build

mikey2020 committed:
Include css file in build/index.html

mikey2020 committed:
Update extract text plugin syntax

mikey2020 committed:
Update webpack.config.prod.js

mikey2020 committed:
Update webpack.config.prod.js

mikey2020 committed:
Update package.json

mikey2020 committed:
Update package.json

mikey2020 committed:
Update build server

mikey2020 committed:
Update build server

mikey2020 committed:
Update procfile to use migrations

mikey2020 committed:
Fix merge conflicts

mikey2020 committed:
Update user model

mikey2020 committed:
Redesign application interface

mikey2020 committed:
Setup application for coverage

mikey2020 committed:
Add istanbul for code coverage

mikey2020 committed:
build dist folder for coverage

mikey2020 committed:
Fix shallow rendering error

mikey2020 committed:
Fix eslint errors

mikey2020 committed:
Fix some eslint errors

mikey2020 committed:
Set up urgent and critical notifications

mikey2020 committed:
Set up email and sms notifications

mikey2020 committed:
Try setting up in-app notifications

mikey2020 committed:
Try setting up in-app notifications

mikey2020 committed:
Group notifications column to group model

mikey2020 committed:
Create notifications component

mikey2020 committed:
feat(client-side): set up message notifications

- Change all .js files to .jsx
- Create notification models
- Create user notifications models
- Create helpers folder in the server directory
- Create socket.js file in server/helpers folder
- Create Notifications.jsx component
- Create notificationActions.js file in client/src/actions directory
- Create notifications.js reducer in client/src/reducers directory

[Delivers #149948430]

mikey2020 committed:
feat(client-side): set up message notifications

- Change all .js files to .jsx
- Create notification models
- Create user notifications models
- Create helpers folder in the server directory
- Create socket.js file in server/helpers folder
- Create Notifications.jsx component
- Create notificationActions.js file in client/src/actions directory
- Create notifications.js reducer in client/src/reducers directory

[Delivers #149948430]

mikey2020 committed:
Merge branch 'develop' of https://github.com/mikey2020/post-it into develop

eick 49f076a Add Heroku generated app.json

mikey2020 committed:
Merge branch 'develop' of https://github.com/mikey2020/post-it into develop

eick 49f076a Add Heroku generated app.json

mikey2020 committed:
Merge branch 'develop' of https://github.com/mikey2020/post-it into develop

mikey2020 committed:
Merge branch 'develop' of https://github.com/mikey2020/post-it into develop

mikey2020 committed:
Merge branch 'develop' of https://github.com/mikey2020/post-it into develop

mikey2020 committed:
Merge branch 'develop' of https://github.com/mikey2020/post-it into develop

mikey2020 committed:
Merge branch 'develop' of https://github.com/mikey2020/post-it into develop

mikey2020 committed:
fix(update readme.md): Update readme.md file

- Re-write readme file to be more descriptive

[Delivers none]

mikey2020 committed:
fix(update readme.md): Update readme.md file

- Re-write readme file to be more descriptive

[Delivers none]

mikey2020 committed:
Merge branch 'develop' of https://github.com/mikey2020/post-it into develop

mikey2020 committed:
Merge branch 'develop' of https://github.com/mikey2020/post-it into develop

mikey2020 committed:
Merge branch 'develop' of https://github.com/mikey2020/post-it into develop

mikey2020 committed:
Merge branch 'develop' of https://github.com/mikey2020/post-it into develop

desaawa commented on client/build.js:
```diff
@@ -0,0 +1,30 @@
+
+process.env.NODE_ENV = 'production';
+
+console.log('Generating minified prod version...'.blue);
```
Let's try and remove all unnecessary `console.log` statements or files. I think this file should not be pushed online

desaawa reviewed (changes requested):
-- I'm getting console errors when I try to log in with wrong credentials, and nothing actually happens, it doesn't throw a nice error message on the modal

-- Phone number should be validated to only accept numbers. I put in a string and it signed me up.
Let's try and ensure that authentication works well, with good validation

foladipo commented:
Hello Michael!

You've done a great job on this project. Your code is clean and lean and you have a great README. I also like the fact that you have separate build scripts for development and for production.

- Add some meat to the description the top of your repo. It's great that it contains a link to an online instance of your app, but a summary of your project will make it even better.

- Add an open source licence to your project, especially in your README and `package.json`. I suggest using the MIT licence.

- I think you should add a `Table of Contents` to your README. This will make it easier to navigate for its readers.

- Install npm packages using `npm install --save` or `npm install --save-dev`, not `npm install --global`. Consequently, you need to change the npm scripts in your `package.json` to use local installations on those packages, not the global ones.

- Fix your ESlint errors and increase your G.P on CodeClimate to ~4.0.

- When describing the type of a parameter in JSDocs, use PascalCase instead of camelCase. In other words, don't use:

```
/**
 * @param {immutableArray} foo - unchangeable arrary
*/
```

Rather, use:

```
/**
 * @param {ImmutableArray} foo - unchangeable arrary
```

- Use routes instead of giving direct access to your controllers.

- Use a layer of route handlers to control access to your controllers, instead of accessing them (controllers) directly. Do this will also make it easier to document your API using Slate, Swagger or any relevant API documentation tool.

- Use an effective folder structure to group components, test files etc.

mikey2020 committed:
Merge branch 'develop' of https://github.com/mikey2020/post-it into develop

mikey2020 committed:
feat(): implement feedback

mikey2020 committed:
chore(): document api

- add /api-docs routes to app.js
- add folder api-docs to root directory

Delivers [#149948560]

mikey2020 committed:
feat(messages): read messages should be archived

- implement pagination in the server side
- set offset and limit in the front end or client side
- move api-docs to client folder
- add a button so when users click they can see old messsages

Delivers [#149948515]

mikey2020 committed:
fix(google-signup): fix google signup

- add a google sign up button on signup form
- set up signup form to set user's google details

Delivers [#150745310]

mikey2020 committed:
feat(forgot-password): implement forgot password feature

- create reset password component
- create two endpoints , one for sending verification code email , another for verifing code

Delivers [#149628797]

mikey2020 committed:
feat(feedback) implement feedback :

 - fix console errors
 - re-design user interface
 - fix failing tests in the client side
 - make tests more descriptive
 - added MIT license to readme
 - wrote more end to end tests
 - add hound.yml file

 Delivers[#150768428]
